database/postgres/postgres_country.py

```diff
@@ -0,0 +1,54 @@
+from typing import Dict
+from database.models import Country
+
+"""def extract_country(data: dict, key: str = "country") -> dict:
+    if not isinstance(data, (dict, list)):
+        return None
+
+    if isinstance(data, dict):
+        if key in data:
+            return data[key]
+        for k, v in data.items():
+            result = extract_country(v, key)
+            if result is not None:
+                return result
+
+    if isinstance(data, list):
+        for item in data:
+            result = extract_country(item, key)
+            if result is not None:
+                return result
+
+    return None"""
+
+def save_country(country_data: Dict, session) -> Country:
+    if not country_data:
+        return None
+
+    if isinstance(country_data, str):
+        country_data = {"name": country_data, "id": None, "code": None, "flag": None}
+
+    country_api_id = country_data.get('id')
+    country_name = country_data.get('name')
+    country_code = country_data.get('code')
+    country_flag = country_data.get('flag')
+
+    if not country_name:
+        return None
+
+    country_entry = session.query(Country).filter_by(name=country_name).first()
```

Comment:
Not sure but isn't there something like "first_or_default" in sqlalchemy?

I'm cautious about this line yielding unwanted exception. Not sure though, will dicuss tomorrow.